subworkflows/nf-core/variantcalling/haplotypecaller/main.nf

```diff
@@ -41,67 +41,69 @@ workflow RUN_HAPLOTYPECALLER {
-    MERGE_HAPLOTYPECALLER(
-        haplotypecaller_vcf_branch.intervals
-            .map{ meta, vcf ->
-
-                new_meta = [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals]
+    if (params.joint_germline) {
+        // merge vcf and tbis
+        genotype_gvcf_to_call = Channel.empty().mix(HAPLOTYPECALLER.out.vcf
+                                .join(HAPLOTYPECALLER.out.tbi)
+                                .join(cram).map{ meta, vcf, tbi, cram, crai, intervals, dragstr_model ->
+                                     [ meta, vcf, tbi, intervals ]
+                                     })
+        // make channels from labels
+            dbsnp_vqsr        = params.dbsnp_vqsr        ? Channel.value(params.dbsnp_vqsr)        : Channel.empty()
+            known_indels_vqsr = params.known_indels_vqsr ? Channel.value(params.known_indels_vqsr) : Channel.empty()
+            known_snps_vqsr   = params.known_snps_vqsr   ? Channel.value(params.known_snps_vqsr)   : Channel.empty()
+
+
+        JOINT_GERMLINE(
+             genotype_gvcf_to_call,
+             fasta,
+             fasta_fai,
+             dict,
+             dbsnp,
+             dbsnp_tbi,
+             dbsnp_vqsr,
+             known_sites_indels,
+             known_sites_indels_tbi,
+             known_indels_vqsr,
+             known_sites_snps,
+             known_sites_snps_tbi,
+             known_snps_vqsr)
+
+        filtered_vcf = JOINT_GERMLINE.out.genotype_vcf.map{ meta, vcf-> [[patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals, variantcaller:"haplotypecaller"], vcf]}
+        ch_versions = ch_versions.mix(JOINT_GERMLINE.out.versions)
+    } else {
+        // Only when using intervals
+        MERGE_HAPLOTYPECALLER(
+            haplotypecaller_vcf_branch.intervals
+                .map{ meta, vcf ->
 
-                [groupKey(new_meta, new_meta.num_intervals), vcf]
-            }.groupTuple(),
-        dict)
+                    new_meta = [patient:meta.patient, sample:meta.sample, status:meta.status, sex:meta.sex, id:meta.sample, num_intervals:meta.num_intervals]
 
-    haplotypecaller_vcf = Channel.empty().mix(
-        MERGE_HAPLOTYPECALLER.out.vcf,
-        haplotypecaller_vcf_branch.no_intervals)
+                    [groupKey(new_meta, new_meta.num_intervals), vcf]
+                }.groupTuple(),
+            dict)
 
-    haplotypecaller_tbi = Channel.empty().mix(
-        MERGE_HAPLOTYPECALLER.out.tbi,
-        haplotypecaller_tbi_branch.no_intervals)
+        haplotypecaller_vcf = Channel.empty().mix(
+            MERGE_HAPLOTYPECALLER.out.vcf,
+            haplotypecaller_vcf_branch.no_intervals)
 
-    if (params.joint_germline) {
+        haplotypecaller_tbi = Channel.empty().mix(
+            MERGE_HAPLOTYPECALLER.out.tbi,
+            haplotypecaller_tbi_branch.no_intervals)
```

Comment:
this should go outside the if clause back up after line 43. Haplotypecaller calls should be published regardless of single sample of joint calling